generator/sbpg/targets/resources/c/new/sbp_messages_template.h

```diff
@@ -41,114 +42,405 @@
-  const char *(((m.name|convert_unpacked)))_(((f.name)))_get(const sbp_(((f.encoding)))_string_t *s);
+  /**
+   * Set (((comment_name)))
+   *
+   * Erase any existing content and replace with the specified string
+   *
+   * This function will return true if the new string was successfully applied. If the specified string is longer than can be represented in wire encoding this function will return false
+   *
+   * @param msg (((msg_type))) instance
+   * @param new_str New string
+   * @return true on success, false otherwise
+   */
+  bool (((prefix)))_set( (((-string_type))) *s, const char *new_str);
+
+  /**
+   * Set (((comment_name))) with printf style formatting
+   *
+   * Erase any existing content and replace with the formatted string
+   *
+   * This function will return true if the new string was successfully applied. If the operation would end up overflowing the maximum size of this field in wire encoding the existing contents will be erased and this function will return false.
+   *
+   * @param msg (((msg_type))) instance
+   * @param fmt printf style format string
+   * @return true on success, false otherwise
+   */
+  bool (((prefix)))_printf( (((-string_type))) *s, const char *fmt, ...) SBP_ATTR_FORMAT(2,3);
+
+  /**
+   * Set (((comment_name))) with printf style formatting
+   *
+   * Identical to #(((prefix)))_printf except it takes a va_list argument
+   *
+   * @param msg (((msg_type))) instance
+   * @param fmt printf style format string
+   * @param ap Argument list
+   * @return true on success, false otherwise
+   */
+  bool (((prefix)))_vprintf( (((-string_type))) *s, const char *fmt, va_list ap);
+
+  /**
+   * Append (((comment_name))) with printf style formatting
+   *
+   * The new string will be appended to the existing contents of the string (if any). If the operation would end up overflowing the maximum size of this field in wire encoding the existing contents will be unmodified and this function will return false.
+   *
+   * @param msg (((msg_type))) instance
+   * @param fmt printf style format string
+   * @return true on success, false otherwise
+   */
+  bool (((prefix)))_append_printf( (((-string_type))) *s, const char *fmt, ...) SBP_ATTR_FORMAT(2,3);
+
+  /**
+   * Append (((comment_name))) with printf style formatting
+   *
+   * Identical to #(((prefix)))_append_printf except it takes a va_list argument
+   *
+   * @param msg (((msg_type))) instance
+   * @param fmt printf style format string
+   * @param ap Argument list
+   * @return true on success, false otherwise
+   *
+   */
+  bool (((prefix)))_append_vprintf( (((-string_type))) *s, const char *fmt, va_list ap);
+
+  /**
+   * Obtain the string value from (((comment_name)))
+   *
+   * @param msg (((msg_type))) instance
+   * @return String contents
+   */
+  const char *(((prefix)))_get(const (((string_type))) *s);
```

Comment:
How will this work for `multipart` or `double_null_terminated` types? for the caller that is?

Reply:
This is in a block which only runs for terminated and null-terminated string. The equivalent function for multipart and double_null_terminated strings is `(((prefix)))_get_section(...)`